### ISSUE TITLE
SNOW-1660331: Should SnowflakeFileTransferOptions.getFilesToStream be exported member?

### ISSUE THREAD

**Issue by shinya-ml** [snowflakedb/gosnowflake#1207]
Please answer these questions before submitting your issue. 
In order to accurately debug the issue this information is required. Thanks!

1. What version of GO driver are you using?

v1.11.1
   
2. What operating system and processor architecture are you using?

macOS 13.2.1
arm64
   
3. What version of GO are you using?
run `go version` in your console

1.21.1

4.Server version:* E.g. 1.90.1
You may get the server version by running a query:
```
SELECT CURRENT_VERSION();
```
8.33.1

5. What did you do?

   If possible, provide a recipe for reproducing the error.
   A complete runnable program is good.

I would like to retrieve the files stored in the internal stage using the GET operation, which was supported in version 1.11.1, and output the contents of the files to the standard output. I referred to the example in the gosnowflake [godocs](https://pkg.go.dev/github.com/snowflakedb/gosnowflake#hdr-Support_For_PUT_and_GET), but when using gosnowflake as an external module, I found that I cannot access the getFilesToStream member of the SnowflakeFileTransferOptions struct.

```golang
package main

import (
	"bytes"
	"compress/gzip"
	"context"
	"database/sql"
	"fmt"
	"io"
	"log"
	"os"

	"github.com/snowflakedb/gosnowflake"
)

func main(){
        dsn, err := gosnowflake.DSN(&gosnowflake.Config{//omitted})
	if err != nil {
		log.Fatal(err)
	}

	db, err := sql.Open(
		"snowflake",
		dsn,
	)
	if err != nil {
		log.Fatal(err)
	}

	if err := db.Ping(); err != nil {
		log.Fatal(err)
	}
	var streamBuf bytes.Buffer
	ctx := gosnowflake.WithFileTransferOptions(context.Background(), &gosnowflake.SnowflakeFileTransferOptions{}) // <-- We cannot access getFilesToStream
	ctx = gosnowflake.WithFileGetStream(ctx, &streamBuf)
	if _, err := db.ExecContext(ctx, "GET @my_stage file://tmp/"); err != nil {
		log.Fatal(err)
	}
	br := bytes.NewReader(streamBuf.Bytes())
	gr, err := gzip.NewReader(br)
	if err != nil {
		log.Fatal(err)
	}
	defer gr.Close()
	io.Copy(os.Stdout, gr)
	fmt.Println()
}
```

6. What did you expect to see?

   What should have happened and what happened instead?

I expected the content of the file obtained via the GET operation to be streamed to the standard output, but the contents of streamBytes only contained EOF. If it is necessary to set getFilesToStream to true, then a means to access it is required.

7. Can you set logging to DEBUG and collect the logs?

   https://community.snowflake.com/s/article/How-to-generate-log-file-on-Snowflake-connectors
   
8. What is your Snowflake account identifier, if any? (Optional)

**Comment by sfc-gh-dszmolka** [comment]
hi and thanks for reporting this issue, taking a look

**Comment by sfc-gh-dszmolka** [comment]
PR in draft https://github.com/snowflakedb/gosnowflake/pull/1208

**Comment by shinya-ml** [comment]
Thank you for investigating the issue. I'm glad we were able to identify issues other than getFilesToStream. I look forward to the resolution of the problems.

**Comment by sfc-gh-dszmolka** [comment]
released with gosnowflake v1.11.2, in September 2024 release cycle
